### PR TITLE
feat(gsd): evidence-grounded prompt contract — unknowns inventory, verification protocol, factcheck reroute

### PR DESCRIPTION
**Pull request by Jamie-BitFlight**
## TL;DR

**What:** Updates GSD prompt templates and dispatch logic to encode evidence-grounded workflow discipline — unknowns inventories, verification protocols, bug-fix protocols, and factcheck-reroute dispatch.
**Why:** Research and execution prompts previously treated recalled claims as facts. This makes unverified claims visible and gives them resolution paths before they become committed code.
**How:** Prompt template changes (markdown), a new `loadFactcheckEvidence()` prompt helper, and a `factcheck-reroute` dispatch rule that reinvokes planning when refuted claims impact slice scope.

### Prompt template changes

Six prompt templates updated to encode evidence discipline:

- **research-milestone.md / research-slice.md** — add required Unknowns Inventory section. Researchers must classify every unverified implementation-affecting claim by evidence basis and resolution path.
- **plan-milestone.md / plan-slice.md** — planners read the unknowns inventory and convert unresolved items into concrete task steps. No silent dropping.
- **execute-task.md** — executors verify claims before acting on them (evidence check step) and follow a structured bug-fix protocol (reproduce → define success → apply → verify).
- **complete-slice.md** — completers report unknowns resolution status (N/M resolved, remaining items, refuted adjustments).

### Research template

- `templates/research.md` — adds the Unknowns Inventory output template with evidence-basis classification fields.

### Dispatch and prompt assembly

- `auto-dispatch.ts` — adds `factcheck-reroute → plan-slice` dispatch rule. When `FACTCHECK-STATUS.json` reports `planImpacting: true`, dispatches `plan-slice` instead of continuing to execution with stale assumptions.
- `auto-prompts.ts` — adds `loadFactcheckEvidence()` which reads refuted claim annotations and injects corrected values into the plan-slice prompt.

## Why

The pipeline had a structural gap: training-data recall was treated as fact. An agent could state `actions/checkout@v4` with confidence and be wrong. The unknowns inventory makes the gap visible. The verification protocol catches it at execution time. The factcheck reroute catches it before execution begins.

## How

- Prompt changes are pure markdown — no runtime behavior change, just different instructions to the LLM
- `loadFactcheckEvidence()` reads from the existing factcheck artifact structure (`factcheck/FACTCHECK-STATUS.json` + `factcheck/claims/*.json`)
- The factcheck-reroute dispatch rule is ordered before the normal `planning → plan-slice` rule so it fires first when evidence exists
- All changes are additive — no existing prompt steps removed, only new steps inserted

## Change type

- [x] `feat` — New feature or capability

## Scope

- [x] `gsd extension` — GSD workflow

## Breaking changes

- [x] No breaking changes

## Test plan

- [ ] CI passes
- [ ] New/updated tests included
- [x] Manual testing — prompt templates are markdown; dispatch rule and prompt helper are covered by existing factcheck test suites on the mega-branch
- [ ] No tests needed — explained above

## AI disclosure

- [x] This PR includes AI-assisted code

### CODE REVIEW

File: src/resources/extensions/gsd/auto-dispatch.ts [hunk 1]
```diff
@@ -268,6 +268,40 @@ const DISPATCH_RULES: DispatchRule[] = [
       };
     },
   },
+  {
+    name: "factcheck-reroute → plan-slice",
+    match: async ({ state, mid, midTitle, basePath }) => {
+      if (state.phase !== "planning") return null;
+      const sid = state.activeSlice!.id;
+      const sTitle = state.activeSlice!.title;
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/auto-dispatch.ts]: Fixed — added null guard for state.activeSlice. Commit 65fe80ad.

File: src/resources/extensions/gsd/prompts/plan-slice.md [hunk 2]
```diff
@@ -63,18 +63,25 @@ Then:
...
+   - **Completion semantics:** If every task were completed exactly as written, the slice goal/demo should actually be true.
+   - **Requirement coverage:** Every must-have in the slice maps to at least one task. No must-have is orphaned. If `REQUIREMENTS.md` exists, every Active requirement this slice owns maps to at least one task.
+   - **Task completeness:** Every task has steps, must-haves, verification, inputs, and expected output — none are blank or vague. Inputs and Expected Output list backtick-wrapped file paths, not prose descriptions.
+   - **Dependency correctness:** Task ordering is consistent. No task references work from a later task.
+   - **Key links planned:** For every pair of artifacts that must connect, there is an explicit step that wires them.
+   - **Scope sanity:** Target 2–5 steps and 3–8 files per task. 10+ steps or 12+ files — must split. Each task must be completable in a single fresh context window.
+   - **Feature completeness:** Every task produces real, user-facing progress — not just internal scaffolding.
+10. If planning produced structural decisions, append them to `.gsd/DECISIONS.md`
+11. {{commitInstruction}}
+12. Update `.gsd/STATE.md`
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/prompts/plan-slice.md]: Fixed — removed STATE.md update step. Commit 65fe80ad.

File: src/resources/extensions/gsd/prompts/plan-milestone.md [hunk 3]
```diff
@@ -48,7 +48,12 @@ Then:
...
-6. If planning produced structural decisions (e.g. slice ordering rationale, technology choices, scope exclusions), append them to `.gsd/DECISIONS.md` (use the **Decisions** output template from the inlined context above if the file doesn't exist yet)
+6. **Read the Unknowns Inventory** from the milestone research doc (if present). For each unresolved unknown:
+   - Map it to the slice where it will be resolved.
+   - If a claim marked `training-data` affects the approach for a slice, note that the slice must verify it.
+   - Do not silently drop unresolved unknowns. Every unresolved item must be assigned to a slice.
+7. If planning produced structural decisions (e.g. slice ordering rationale, technology choices, scope exclusions), append them to `.gsd/DECISIONS.md` (use the **Decisions** output template from the inlined context above if the file doesn't exist yet)
+8. Update `.gsd/STATE.md`
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/prompts/plan-milestone.md]: Fixed — removed STATE.md update step. Commit 65fe80ad.

File: src/resources/extensions/gsd/prompts/execute-task.md [hunk 4]
```diff
@@ -58,13 +69,14 @@ Then:
...
+15. If you made an architectural, pattern, library, or observability decision during this task that downstream work should know about, append it to `.gsd/DECISIONS.md` (use the **Decisions** output template from the inlined templates below if the file doesn't exist yet). Not every task produces decisions — only append when a meaningful choice was made.
+16. If you discover a non-obvious rule, recurring gotcha, or useful pattern during execution, append it to `.gsd/KNOWLEDGE.md`. Only add entries that would save future agents from repeating your investigation. Don't add obvious things.
+17. Use the **Task Summary** output template from the inlined templates below
+18. Write `{{taskSummaryPath}}`
+19. Mark {{taskId}} done in `{{planPath}}` (change `[ ]` to `[x]`)
+20. Do not run git commands — the system auto-commits your changes after this unit completes.
+21. Update `.gsd/STATE.md`
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/prompts/execute-task.md:109]: Fixed — removed STATE.md update step. Commit 65fe80ad.

File: src/resources/extensions/gsd/prompts/complete-slice.md [hunk 5]
```diff
@@ -24,13 +24,19 @@ Then:
...
+   - If unknowns remain unresolved, note what was attempted and why resolution failed.
+7. Write `{{sliceSummaryPath}}` (compress all task summaries).
+8. Write `{{sliceUatPath}}` — a concrete UAT script with real test cases derived from the slice plan and task summaries. Include preconditions, numbered steps with expected outcomes, and edge cases. This must NOT be a placeholder or generic template — tailor every test case to what this slice actually built.
+9. Review task summaries for `key_decisions`. Append any significant decisions to `.gsd/DECISIONS.md` if missing.
+10. Review task summaries for patterns, gotchas, or non-obvious lessons learned. If any would save future agents from repeating investigation or hitting the same issues, append them to `.gsd/KNOWLEDGE.md`. Only add entries that are genuinely useful — don't pad with obvious observations.
+11. Mark {{sliceId}} done in `{{roadmapPath}}` (change `[ ]` to `[x]`)
+12. Do not run git commands — the system auto-commits your changes and handles the merge after this unit succeeds.
+13. Update `.gsd/PROJECT.md` if it exists — refresh current state if needed.
+14. Update `.gsd/STATE.md`
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/prompts/complete-slice.md:43]: Fixed — removed STATE.md update step. Commit 65fe80ad.

File: src/resources/extensions/gsd/auto-prompts.ts [hunk 6]
```diff
@@ -465,6 +465,111 @@ export function extractSliceExecutionExcerpt(content: string | null, relPath: st
...
+
+  try {
+    const statusContent = await loadFile(factcheckStatusPath);
+    if (!statusContent) return null;
+    const status = JSON.parse(statusContent);
+
+    // Only inject when there are refutations that impact the plan
+    if (status.overallStatus !== "has-refutations") return null;
+    if (!status.planImpactingClaims || status.planImpactingClaims.length === 0) return null;
+
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/auto-prompts.ts:639]: Fixed — gate now accepts both overallStatus and planImpacting. Commit 65fe80ad.

File: src/resources/extensions/gsd/auto-prompts.ts [hunk 7]
```diff
@@ -465,6 +465,111 @@ export function extractSliceExecutionExcerpt(content: string | null, relPath: st
...
+      const claimPath = join(claimsDir, `${claimId}.json`);
+      if (!existsSync(claimPath)) continue;
+      try {
+        const claimContent = await loadFile(claimPath);
+        if (!claimContent) continue;
+        const claim = JSON.parse(claimContent);
+        // Only include REFUTED claims with corrected values
+        if (claim.verdict === "refuted" && claim.correctedValue) {
+          refutedClaims.push({
+            claimId: claim.claimId,
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/auto-prompts.ts]: Fixed — claimId falls back to loop variable. Commit 65fe80ad.

File: src/resources/extensions/gsd/auto-dispatch.ts [hunk 8]
```diff
@@ -268,6 +268,40 @@ const DISPATCH_RULES: DispatchRule[] = [
...
+
+      // Dispatch plan-slice with factcheck evidence (prompt builder handles injection)
+      return {
+        action: "dispatch",
+        unitType: "plan-slice",
+        unitId: `${mid}/${sid}`,
+        prompt: await buildPlanSlicePrompt(mid, midTitle, sid, sTitle, basePath),
+      };
+    },
+  },
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/auto-dispatch.ts:338]: Test coverage for this dispatch rule is provided by the factcheck test suite in PR #1909.

File: src/resources/extensions/gsd/auto-prompts.ts [hunk 9]
```diff
@@ -465,6 +465,111 @@ export function extractSliceExecutionExcerpt(content: string | null, relPath: st
...
+      lines.push(`- Reroute target: ${status.rerouteTarget}`);
+    }
+    lines.push("");
+    lines.push("**Action Required:** Incorporate the corrected values above into your plan. Do not rely on the refuted claims.");
+
+    return lines.join("\n");
+  } catch {
+    return null;
+  }
+}
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/auto-prompts.ts:714]: Test coverage for loadFactcheckEvidence() is provided by the factcheck test suite in PR #1909.

File: src/resources/extensions/gsd/templates/research.md [hunk 10]
```diff
@@ -77,3 +77,32 @@
...
+       read-code      — read actual source files in this project
+       experiment     — run a command, test an API, try the thing
+       ask-user       — present findings, ask for confirmation or input
+       fetch-reference — retrieve external resource (screenshot, clone repo, scrape)
+       search         — web search for current information
+-->
+
+| # | Claim | Basis | Affects | Resolution | Status |
+|---|-------|-------|---------|------------|--------|
+| {{N}} | {{specificClaim}} | {{basis}} | {{whatDecisionThisAffects}} | {{strategyType}}: {{specificAction}} | unresolved |
```
[the remark this answers is not in the document]
Comment (Jamie-BitFlight) [reply src/resources/extensions/gsd/templates/research.md:108]: Checked — table uses single | delimiters. The || appearance is a diff rendering artifact.

**Review by trek-e — CHANGES_REQUESTED** [review]
**REQUEST CHANGES** -- Part 1 of a 3-PR series. Has unresolved merge conflicts.

This PR introduces the evidence-grounded prompt contract: unknowns inventory sections in research prompts, verification protocol in execute-task prompts, and fact-check reroute dispatch logic.

**Critical issue: Merge conflicts in prompt files.**

The diff shows unresolved git conflict markers in at least two files:

- **complete-slice.md:** Contains \`<<<<<<< HEAD\`, \`=======\`, and \`>>>>>>> 8524093e\` / \`>>>>>>> 65fe80ad\` markers. The prompt has duplicate instruction sets from two different merge attempts that were never resolved.

- **execute-task.md:** Same conflict markers present. Steps are duplicated with different numbering (steps 1-13 from HEAD, steps 1-21 from the feature branch), with conflict markers visible in the raw text.

These conflict markers will be injected verbatim into agent prompts at runtime, causing unpredictable agent behavior.

**Other concerns:**

1. **getLoadedSkills stub:** \`const getLoadedSkills = (() => []) as any;\` replaces the real import. This silently disables all skill loading in auto-prompts.ts. If the import was removed upstream, this needs a proper resolution -- not a stub that breaks skill activation.

2. **loadFactcheckEvidence (auto-prompts.ts, 105 lines):** Reads and formats fact-check claim files from the slice's factcheck directory. The function is well-structured but the error handling silently swallows parse failures. Consider logging a warning on malformed FACTCHECK-STATUS.json so users know their fact-check artifacts are corrupt.

3. **Prompt changes are substantive.** The unknowns inventory requirement in research prompts (research-milestone.md, research-slice.md) and the evidence-check protocol in execute-task.md are meaningful behavioral changes to how agents operate. These deserve careful evaluation by the maintainer.

4. **auto-dispatch.ts factcheck-reroute rule:** The dispatch rule that reroutes to plan-slice when FACTCHECK-STATUS.json has planImpacting=true is clean. The guard clauses (phase check, slice existence, file existence, JSON parse) are correct.

**Blocking issue:** Resolve the merge conflicts in complete-slice.md and execute-task.md before this can be reviewed further. Then address the getLoadedSkills stub. This PR, along with #1908 and #1909, should be reviewed as a unified body of work.

**Comment by trek-e** [comment]
Thank you for your contribution! Unfortunately, our codebase is evolving rapidly and since we haven't received a response to the requested changes within 5 days, we're closing this PR. You're very welcome to rebase your changes against a more current release and resubmit — we'd love to see this work land. Thanks again!
